### PR TITLE
Add 'dimension' output to filters.sample

### PR DESCRIPTION
**Pull request by hobu**
If the user sets a `dimension` option on `filters.sample`, a Uint8 dimension with that name will be populated 1 or 0 whether or not the point was sampled instead of culling points.

This PR also added the ability to set the voxelization origin explicitly with `origin_x`, `origin_y`, and `origin_z` options.

### CODE REVIEW

File: doc/stages/filters.sample.rst [hunk 1]
```diff
@@ -73,6 +73,20 @@ cell
   Voxel cell size. If ``radius`` is set, ``cell`` is automatically computed
   such that the cell is circumscribed by the sphere defined by ``radius``.
 
+dimension
```
Comment (abellgithub) [inline doc/stages/filters.sample.rst:76]: Not sure I love this name. Maybe `indicator_dim` or some such? Names are hard.

File: filters/SampleFilter.cpp [hunk 2]
```diff
@@ -90,8 +103,24 @@ PointViewSet SampleFilter::run(PointViewPtr view)
...
+
             output->appendPoint(*view, point.pointId());
+        } else
+        {
+            // If the user set a dimension, write 0s be cause we didn't
+            // keep the point. Otherwise we drop it on the floor.
+            if (m_dimension != pdal::Dimension::Id::Unknown)
+            {
+                point.setField(m_dimension, 0);
+                output->appendPoint(*view, point.pointId());
```
Comment (abellgithub) [inline filters/SampleFilter.cpp:121]: `appendPoint` is incorrect here.

===

You could use this litter helper or move its logic into `voxelize`

```
bool func(point)
{
  bool keep = voxelize(point);
  if (m_dimension != pdal::Dimension::Id::Unknown)
  {
    point.setField(m_dimension, keep);
    keep = true;
  }
  return keep;
}
```
In run():

```
if (func(point))
  output->appendPoint(*view, point.pointId());
``` 

In processOne():

```
return func(point);
```

**Review by abellgithub — CHANGES_REQUESTED** [review]
Bug needs fixing.

File: doc/stages/filters.sample.rst (hunk 1 above)
Comment (hobu) [reply doc/stages/filters.sample.rst:76]: We use this name in `filters.overlay` already, so I'm just following suit.

File: filters/SampleFilter.cpp (hunk 2 above)
Comment (hobu) [reply filters/SampleFilter.cpp]: updated as suggested

**Pull request merged**